crates/sqlbuiltins/src/functions/scalars/sentence_transformers.rs

```diff
@@ -0,0 +1,253 @@
+use std::sync::Arc;
+
+use candle_core::{CpuStorage, Device, Storage, Tensor, WithDType};
+use candle_nn::VarBuilder;
+use candle_transformers::models::bert::{BertModel, Config, DTYPE};
+use datafusion::arrow::array::{Array, AsArray, FixedSizeListBuilder, Float32Builder};
+use datafusion::arrow::datatypes::{DataType, Field};
+use datafusion::error::DataFusionError;
+use datafusion::logical_expr::expr::ScalarFunction;
+use datafusion::logical_expr::{
+    Expr,
+    ReturnTypeFunction,
+    ScalarFunctionImplementation,
+    ScalarUDF,
+    ScalarUDFImpl,
+    Signature,
+    Volatility,
+};
+use datafusion::physical_plan::ColumnarValue;
+use datafusion::scalar::ScalarValue;
+use hf_hub::api::sync::Api;
+use hf_hub::{Repo, RepoType};
+use protogen::metastore::types::catalog::FunctionType;
+use tokenizers::{PaddingParams, Tokenizer};
+
+use crate::functions::{BuiltinScalarUDF, ConstBuiltinFunction, FunctionNamespace};
+
+#[derive(Clone)]
+pub struct BertUdf {
+    model: Arc<BertModel>,
+    signature: Signature,
+    tokenizer: Tokenizer,
+    device: Device,
+    output_dim: usize,
+}
+
+impl std::fmt::Debug for BertUdf {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("BertUdf").finish()
+    }
+}
+
+impl BertUdf {
+    pub fn new() -> Self {
+        let device = Device::Cpu;
+        let model_id = "sentence-transformers/all-MiniLM-L6-v2".to_string();
```

Comment:
does this make sense to put in a constant somewhere?

Reply:
There are many Bert models, The idea was to have that as a kind of base struct that each sentence_transformer udf would have their own static implementation of. So the config vars like this will eventually be a part of that.